optuna/trial.py

```diff
@@ -109,6 +109,12 @@ def datetime_start(self):
 
         raise NotImplementedError
 
+    @property
+    def number(self):
+        # type: () -> int
```

Comment:
Optuna dropped Python2 support and we can use a function annotation now. Please use it instead of the type comment. (See https://github.com/optuna/optuna/issues/711)
```suggestion
    def number(self) -> int:
```

Reply:
Thank you for pointing it out. I resolved it.